src/NuGet.Services.EndToEnd/IconTests.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Services.EndToEnd.Support;
+using NuGet.Services.EndToEnd.Support.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NuGet.Services.EndToEnd
+{
+    [Collection(nameof(PushedPackagesCollection))]
+    public class IconTests : IClassFixture<TrustedHttpsCertificatesFixture>
+    {
+        private readonly PushedPackagesFixture _pushedPackages;
+        private readonly Clients _clients;
+        private readonly ITestOutputHelper _logger;
+
+        public IconTests(PushedPackagesFixture pushedPackages, ITestOutputHelper logger)
+        {
+            _pushedPackages = pushedPackages;
+            _clients = pushedPackages.Clients;
+            _logger = logger;
+        }
+
+        [Theory]
+        [InlineData(PackageType.EmbeddedIconJpeg)]
+        [InlineData(PackageType.EmbeddedIconPng)]
+        public async Task PushedPackageWithIconIsAvailableInV3(PackageType packageType)
+        {
+            string expectedExtension;
+            switch (packageType)
+            {
+                case PackageType.EmbeddedIconJpeg:
+                    expectedExtension = ".jpg";
+                    break;
+
+                case PackageType.EmbeddedIconPng:
+                    expectedExtension = ".png";
+                    break;
+
+                default:
+                    throw new ArgumentException($"Unsupported package type {packageType}");
+            }
+            var expectedFilename = $"icon{expectedExtension}";
```

Comment:
This is really "input extension" right? The URL in our endpoints has no file extension I thought